### ISSUE TITLE
[Question] Are there plans to support a sparse record collection?

### ISSUE THREAD

**Issue by pixelhandler** [emberjs/ember-collection#47]
At a quick scan of the README I see the example that ember-collection will mange rendering 10,000 items and only draw what should be on screen. 

My curiosity is… Since I will never ask my API for 10K records, does ember-collection also mange loading records in chucks?

Are there any plans for supporting a sparse array of models? And only load what should be loaded and drawn accordingly?

**Comment by stefanpenner** [comment]
> My curiosity is… Since I will never ask my API for 10K records, does ember-collection also mange loading records in chucks?

typically I have accomplished this successfully at a layer above ember-collection.

**Comment by raytiley** [comment]
So there is some discussion about it here: https://github.com/emberjs/ember-collection/pull/27 but essentially the idea is send an action when the visible content changes, so you can load more content as needed.

**Comment by stefanpenner** [comment]
> but essentially the idea is send an action when the visible content changes, so you can load more content as needed.

yes, this sort of protocol i believe is what we we should encourage.

**Comment by lukemelia** [comment]
Yes, @krisselden has an idea for a datasource API that will facilitate sparse array support.
